Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,70 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
+    # TODO: goal is to no longer require def commands at all
+    CommandsCommand.call
+  end
+
+  class CommandsCommand < Command
+    options do
+      command "commands"
+      desc "Show a list of built-in and external commands."
+      option "--quiet", "list only the names of commands without the header." do
+        option "--include-aliases", "the aliases of internal commands will be included."
+      end
+    end
+
+    def self.call
+      # TODO: Put this check_invalid_options() method such that it doesn't have to be here
+      check_invalid_options
+
+      if argv.include? "--quiet"
```

Comment:
I think this should be a bit of a nicer interface than just checking for array inclusion like `argv`. Either turning this into a Hash or even methods for each option would be nice. Additionally, I wouldn't name this `argv` as that means it looks a lot like it's just `ARGV`. Instead, a global, immutable option like `PassedOptions` or something (that names not great so feel free to change it).